stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/controller/ChannelControllerImpl.kt

```diff
@@ -608,26 +630,43 @@ internal class ChannelControllerImpl(
 
         return if (online) {
             // upload attachments
-            logger.logI("Uploading attachments for message with id ${newMessage.id} and text ${newMessage.text}")
-            newMessage.attachments = newMessage.attachments.map {
-                var attachment: Attachment = it
-                if (it.upload != null) {
-                    val result = uploadAttachment(it, attachmentTransformer)
-                    attachment.uploadComplete = result.isSuccess
-                    if (result.isSuccess) {
-                        attachment = result.data()
-                    } else {
-                        attachment.uploadState = Attachment.UploadState.Failed(result.error())
-                        logger.logE("Failed to upload attachment for message")
+
+            if (hasAttachments) {
+                logger.logI("Uploading attachments for message with id ${newMessage.id} and text ${newMessage.text}")
+
+                newMessage.attachments = newMessage.attachments.mapIndexed { i, attach ->
+                    var attachment: Attachment = attach
+                    val attachmentProgress = attachmentProgressList[i]
+
+                    if (attachment.upload != null) {
+                        val result = uploadAttachment(
+                            attachment,
+                            attachmentTransformer,
+                            attachmentProgress.toProgressCallback()
+                        )
+
+                        if (result.isSuccess) {
+                            attachment = result.data()
+                            attachmentProgress.setComplete(true)
+                            attachment.uploadState = Attachment.UploadState.Success
+                        } else {
+                            attachmentProgress.setComplete(false)
+                            attachment.uploadState = Attachment.UploadState.Failed(result.error())
+                        }
                     }
-                }
-                attachment
-            }.toMutableList()
 
-            logger.logI("Starting to send message with id ${newMessage.id} and text ${newMessage.text}")
+                    attachment
+                }.toMutableList()
 
+                uploadStatusMessage?.let { cancelMessage(it) }
+                uploadStatusMessage = null
```

Comment:
This is inside the `if (online)` condition. If we're not online, this property is never cleared?

Reply:
Yes, but this is an ephemeral message. The user will see this message, but no update in the progress of the attachment. If the user leave the app, the message is not persistet

Reply:
The current behaviour is that the message with the attachment stays in the conversation if the user is not online

Reply:
Yep, checked.

Reply:
Made `uploadStatusMessage = null` in the offline mode as we discussed.